apps/zui/src/core/electron-zed-client.ts

```diff
@@ -0,0 +1,7 @@
+import {Client} from "@brimdata/zed-node"
+import {net} from "electron"
+
+export class ElectronZedClient extends Client {
+  fetch =
+    process.env.JEST_WORKER_ID === undefined ? net.fetch : globalThis.fetch
```

Comment:
I mentioned in a previous comment that despite the app functionality and e2e tests working ok after the changes thus far, we were still seeing many failures during `yarn test` that looked like the following, such as in [this CI run](https://github.com/brimdata/zui/actions/runs/8992924390):

```
FAIL apps/zui/src/electron/start.test.ts
  ● app opens a window on startup

    TypeError: Cannot read properties of undefined (reading 'fetch')

      3 |
      4 | export class ElectronZedLake extends Lake {
    > 5 |   fetch = net.fetch
        |               ^
      6 | }
      7 |

      at new fetch (src/core/electron-zed-lake.ts:5:15)
      at Function.boot (src/core/main/main-object.ts:47:18)
      at Object.boot (src/electron/run-main/boot.ts:13:16)
      at Object.main (src/electron/run-main/run-main.ts:19:19)
      at Object.<anonymous> (src/electron/start.test.ts:18:20)

  ● activate creates window if there are none
```

Based on my understanding, these Jest tests are more lightweight so I assume the root cause is that Electron stuff like `net.fetch` are simply not available when Jest tests are running. I figured one way to address this would be to determine if a test was running in Jest and, if so, choose a different `fetch`. I did some web searches and saw others claiming success with checking Jest-ness via the presence of this `JEST_WORKER_ID` env var so I used that approach here and it does seem to work, as `yarn test` now passes locally and in CI and the e2e tests still pass. I'd appreciate some close scrutiny of my approach to confirm it's sound, though!